iceoryx_posh/test/moduletests/test_mepoo_segment_management.cpp

```diff
@@ -21,7 +21,7 @@
 #include "iceoryx_utils/error_handling/error_handling.hpp"
 #include "iceoryx_utils/internal/posix_wrapper/shared_memory_object.hpp"
 #include "iceoryx_utils/internal/posix_wrapper/shared_memory_object/allocator.hpp"
-#include "iceoryx_utils/internal/relocatable_pointer/relative_ptr.hpp"
+#include "iceoryx_utils/internal/relocatable_pointer/base_relative_ptr.hpp"
```

Comment:
We do not need to change it now, but now hat the base version is in separate header I would prefer we would never have to include it since it is an implementation detail.  relative.ptr.hpp alone should define the interface.
This would require more refactoring though to e.g. have all those static methods and constants in the rp namespace.

I would prefer this not to be the last refactoring but it is ok for now.

Reply:
added to issue